agent/engine/dependencygraph/graph_test.go

```diff
@@ -444,3 +444,90 @@ func assertResolved(f func(target *api.Container, dep *api.Container) bool, targ
 		assert.Equal(t, expectedResolved, resolved)
 	}
 }
+
+func TestTransitionDependenciesResolved(t *testing.T) {
+	testcases := []struct {
+		Name             string
+		TargetKnown      api.ContainerStatus
+		TargetDesired    api.ContainerStatus
+		DependencyKnown  api.ContainerStatus
+		DependentStatus  api.ContainerStatus
+		SatisfiedStatus  api.ContainerStatus
+		ExpectedResolved bool
+	}{
+		{
+			Name:             "Nothing running, pull depends on running",
+			TargetKnown:      api.ContainerStatusNone,
+			TargetDesired:    api.ContainerRunning,
+			DependencyKnown:  api.ContainerStatusNone,
+			DependentStatus:  api.ContainerPulled,
+			SatisfiedStatus:  api.ContainerRunning,
+			ExpectedResolved: false,
+		},
+		{
+			Name:             "Nothing running, create depends on running",
+			TargetKnown:      api.ContainerStatusNone,
+			TargetDesired:    api.ContainerRunning,
+			DependencyKnown:  api.ContainerStatusNone,
+			DependentStatus:  api.ContainerCreated,
+			SatisfiedStatus:  api.ContainerRunning,
+			ExpectedResolved: true,
+		},
+		{
+			Name:             "Dependency created, pull depends on running",
+			TargetKnown:      api.ContainerStatusNone,
+			TargetDesired:    api.ContainerRunning,
+			DependencyKnown:  api.ContainerCreated,
+			DependentStatus:  api.ContainerPulled,
+			SatisfiedStatus:  api.ContainerRunning,
+			ExpectedResolved: false,
+		},
+		{
+			Name:             "Dependency running, pull depends on running",
+			TargetKnown:      api.ContainerStatusNone,
+			TargetDesired:    api.ContainerRunning,
+			DependencyKnown:  api.ContainerRunning,
+			DependentStatus:  api.ContainerPulled,
+			SatisfiedStatus:  api.ContainerRunning,
+			ExpectedResolved: true,
+		},
+		{
+			Name:             "Dependency running, create depends on created",
+			TargetKnown:      api.ContainerPulled,
+			TargetDesired:    api.ContainerRunning,
+			DependencyKnown:  api.ContainerRunning,
+			DependentStatus:  api.ContainerCreated,
+			SatisfiedStatus:  api.ContainerCreated,
+			ExpectedResolved: true,
+		},
+		{
```

Comment:
Really appreciate these test cases. Can you also add some test cases for `SatisfiedStatus` being `api.ContainerResourcesReady`?

Reply:
Done